### ISSUE TITLE
Database Upgrade Failure

### ISSUE THREAD

**Issue by praeluceo** [Tautulli/Tautulli#910]
![jira-current1](https://cloud.githubusercontent.com/assets/335284/20688017/d1075c00-b573-11e6-9d9b-c7479e2dee81.png)

**PLEXPY CONFIGURATION**

Git Branch:	master
Git Commit Hash:	3742f33d08feb6fb15e74af6318185230205006f
Configuration File:	/opt/plexpy/config.ini
Database File:	/opt/plexpy/plexpy.db
Log File:	/opt/plexpy/logs/plexpy.log
Backup Directory:	/opt/plexpy/backups
Cache Directory:	/opt/plexpy/cache
GeoLite2 Database:	/opt/plexpy/GeoLite2-City.mmdb | Reinstall / Update | Uninstall
Arguments:	['--quiet', '--daemon', '--nolaunch', '--pidfile=/var/run/plexpy/plexpy.pid', '--datadir=/opt/plexpy']
Platform:	Linux 3.19.0-74-generic
Python Version:	2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

**What you did?**
Performed a regular git upgrade or experienced a power outage.
**What happened?**
"1,968 Days" of playback was recorded at that point.
**What you expected?**
No changes to the database recording playback.

**How can we reproduce your issue?**
I have not attempted a reproduction, I didn't notice the issue until after the fact. We had some bad weather which caused my Plex & PlexPy server to be rebooted a couple times, and I have performed a couple of PlexPy git upgrades, either of which might've happened on the day when the error occurred.

 When I check the actual playback history from that day, I can't see any spurious entries that would relate to the clutter, there are only 10 historical entries on 11/12/2016 in the "Movies" category, and none show duplicate plays, or extremely long duration numbers. I was hoping another upgrade might address an already reported issue, but it didn't. I am considering running "Extra Settings| Flush" to see if that helps, otherwise I'll have to try my hand at SQL and delete the errant data, or copy over a backed up database from /opt/plexpy/backups  (although I haven't checked if that is automatic yet).

**What are your (relevant) settings?**
None of my current settings seem relevant to the issue, except maybe the log rotation schedule, which I should try to find.

**Link to logs:**
My PlexPy logs don't go back more than 1,000 entries, and so don't show the event that did this.

**Comment by JonnyWong16** [comment]
https://www.reddit.com/r/PleX/comments/59eu82/plexpy_insane_amount_of_concurrent_streams_bug/d98nd3z/

**Comment by praeluceo** [comment]
Yes, that does look identical! I'll try the fix in the reddit post when I get home from work.

**Comment by praeluceo** [comment]
I apologize for the delay in getting back to you, I executed the steps from the reddit post you linked to:

> $ /etc/init.d/plexpy stop
> $ cp plexpy.db plexpy.2012-12-12.db
> $ sqlite3 /opt/plexpy/plexpy.db
> sqlite> DELETE FROM session_history WHERE id NOT IN (SELECT id FROM session_history_metadata);
> sqlite> DELETE FROM session_history_media_info WHERE id NOT IN (SELECT id FROM session_history_metadata);
> sqlite> .quit
>$ /etc/init.d/plexpy start

![plexpy_working](https://cloud.githubusercontent.com/assets/335284/21129322/aa1679c6-c0b5-11e6-8ee6-6c33088ad2c6.png)

And the statistics all show up correctly now! Thank you, closing issue (if that's okay with you), although perhaps a feature to detect substantial outliers like this, or a "purge extraneous data" button under settings which restarts plexpy and deletes the extra orphaned entries would be helpful?

Anyway, thanks again for the quick reply. Closing issue.
